gensim/models/word2vec.py

```diff
@@ -320,14 +315,14 @@ def score_sg_pair(model, word, word2):
     l1 = model.syn0[word2.index]
     l2a = deepcopy(model.syn1[word.point])  # 2d matrix, codelen x layer1_size
     sgn = (-1.0)**word.code  # ch function, 0-> 1, 1 -> -1
-    lprob = -log(1.0 + exp(-sgn*dot(l1, l2a.T)))
+    lprob = log(expit(sgn * dot(l1, l2a.T)))
     return sum(lprob)
 
 
 def score_cbow_pair(model, word, word2_indices, l1):
     l2a = model.syn1[word.point]  # 2d matrix, codelen x layer1_size
     sgn = (-1.0)**word.code  # ch function, 0-> 1, 1 -> -1
-    lprob = -log(1.0 + exp(-sgn*dot(l1, l2a.T)))
+    lprob = log(expit(sgn * dot(l1, l2a.T)))
```

Comment:
Not directly related to this PR, but since we're on the topic, there's a function in numpy for safely adding up logs of exps: `numpy.logaddexp(0, sgn * dot(l1, l2a.T))`.